-Added difficulty level buttons nested within 1P mode button.

Platnumrcn14 opened the pull request:
**homeScreen.html**
-Implemented a difficulty selector option nested in the 1P mode button of the main menu
![2024-09-25](https://github.com/user-attachments/assets/2e3fe4d7-9797-4951-b539-414e2542ff02)
 
![2024-09-25 (6)](https://github.com/user-attachments/assets/5f15e61c-a52d-4ce7-9d61-d84fa00054ed)

**server.js**
-created the framework for homeScreen.html to communicate the difficulty option selected to the server.js
![2024-09-25 (1)](https://github.com/user-attachments/assets/3d8d749c-6ab0-4b5e-a5be-6ace8f1cbc67)

**battleshipRound.js**
-Implemented Hard mode for AI that will iterate through the game board for a hittable grid position. Once found will hit that position. 
-Medium mode nested within easy mode
![2024-09-25 (5)](https://github.com/user-attachments/assets/2aa64f00-f6aa-4b96-a1b2-c728f4faeaed)
![2024-09-25 (7)](https://github.com/user-attachments/assets/031142ce-4d54-404f-b29f-a2e57cb018f2)

Platnumrcn14 committed:
-Added difficulty level buttons nested within 1P mode button.
-Implemented Hard mode for AI that will iterate through the game board for a hittable grid position. Once found will hit that position.
-Medium mode nested within easy mode

jacobwilki01 commented on back/server/server.js:
```diff
@@ -91,9 +91,17 @@ io.on('connection', (socket) => {
     // Placeholder for future AI implementation
-    socket.on('initiateSinglePlayer', (numShips) => {
+    socket.on('initiateSinglePlayer', (numShips, difficulty) => {
         // TODO: ALLOW SPECIFYING DIFFICULTY
```
remove todo comment

jacobwilki01 commented on front/homeScreen/homeScreen.html:
why'd you change the spacing of this file so much?

jacobwilki01 commented on back/server/modules/battleshipRound.js:
```diff
@@ -180,6 +186,17 @@ class BattleshipRound {
         return {x:x, y:y}
     }
 
+    targetNextShipTile(opMap) {
```
this is an inefficient way of doing this. Your current implementation has us iterate through the grid **every single time** its the AI's turn, and checks tiles we know are hit already.

Instead, lets generate a list of "tiles to hit" at the start of the first AI turn that is stored in this class. Then, whenever its the AI's turn, grab the next item in the list, remove it from the list, and return that tile.

eBerkley reviewed (approved):
Looks good to me.

Platnumrcn14 committed:
-Modified function used in Hard mode to scan the user's board only once, applying found coodinates to a list.

Platnumrcn14 committed:
-Removed Todo comment and placeholder comment

jacobwilki01 commented on back/server/modules/battleshipRound.js:
```diff
@@ -180,6 +187,27 @@ class BattleshipRound {
     }
 
+    targetShipTiles(opMap) {
+        // If the target list is empty, we need to scan the grid and populate it
```
Let's invert the order of these. So:

if (this.aiTargetList.Length > 0) {
  return this.aiTargetList.shift();
}

THEN generate the list and return the first value without the if statement. That removes the need for the length check of 0 and some of the nesting.

Platnumrcn14 committed:
-Resolved hard mode logic flow beginning at checking if ship positions list has been made, if not creating a list of ship positions

Platnumrcn14 committed:
-Implemented some medium difficulty logic that will target the surrounding squares once a hit is achieved.
-Fixed a bug where the ai would not target row 10 or column 10

Platnumrcn14 committed:
-Completed medium difficulty logic
         -medium logic attempts to destroy ship by attacking in line one direction at a time. switches direction if miss is encountered
-Reorganized definitions within BattleshipRound to refelect order used in aiTurn

*optional future improvement to implement*
-randomizing the directions attacked once first hit is scored

modmo22 reviewed (approved):
Looks great!
